core/src/Threads/Kokkos_ThreadsExec.cpp

```diff
@@ -796,22 +799,7 @@ int g_threads_space_factory_initialized =
     initialize_space_factory<ThreadsSpaceInitializer>("050_Threads");
 
 void ThreadsSpaceInitializer::initialize(const InitArguments &args) {
-  const int num_threads = args.num_threads;
-  if (std::is_same<Kokkos::Threads, Kokkos::DefaultExecutionSpace>::value ||
-      std::is_same<Kokkos::Threads,
-                   Kokkos::HostSpace::execution_space>::value) {
```

Comment:
I fail to see how that would not be always true and definitely would not want to silently do nothing anyway